packages/web-components/src/components/va-date/va-date.tsx

```diff
@@ -52,6 +79,39 @@ export class VaDate {
   dateBlur: EventEmitter;
 
   private handleDateBlur = (event: FocusEvent) => {
+    const [year, month, day] = (this.value || '').split('-').map(val => val);
+    const yearNum = parseInt(year);
+    const monthNum = parseInt(month);
+    const dayNum = parseInt(day);
+    // Use a leading zero for numbers < 10
+    const numFormatter = new Intl.NumberFormat('en-US', {
+      minimumIntegerDigits: 2,
+    });
+    this.value = `${year}-${month ? numFormatter.format(monthNum) : ''}-${
+      day ? numFormatter.format(dayNum) : ''
+    }`;
+    const daysForSelectedMonth = monthNum > 0 ? days[monthNum] : [];
+    const leapYear = checkLeapYear(yearNum);
+    // Check validity of date if invalid provide message and error state styling
+    if (
+      yearNum < minYear ||
+      yearNum > maxYear ||
+      monthNum < minMonths ||
+      monthNum > maxMonths ||
+      dayNum < minMonths ||
+      dayNum > daysForSelectedMonth.length ||
+      day === '' ||
+      month === '' ||
+      year === '' ||
+      (!leapYear && monthNum === 2 && dayNum > 28) ||
+      (this.required && !isFullDate(this.value))
+    ) {
+      this.error = 'Please enter a valid date';
+    } else if (this.error !== 'Please enter a valid date') {
+      this.error;
```

Comment:
What's this doing?

Reply:
It fixes the issue we were running into with someone setting the error prop elsewhere crashing with baked in validation that was causing me to have to add in that `invalid` prop originally. So now if someone uses custom validation like the final example in SB and then we also have default validation elsewhere they won't override each other